src/countries/countries.controller.ts

```diff
@@ -1,7 +1,50 @@
 import { CountriesService } from './countries.service';
-import { Controller } from '@nestjs/common';
+import { Body, Controller, Delete, Get, Param, Post, Put, Req } from "@nestjs/common";
+import { CreateCountryDto } from './dto/create-country.dto';
+import { CountryInterface } from './country.interface';
+import { UpdateCountryDto } from './dto/update-country.dto';
+import { DeleteResult, UpdateResult } from 'typeorm';
 
-@Controller()
+@Controller('countries')
 export class CountriesController {
   constructor(private readonly countryService: CountriesService) {}
+
+  @Get()
+  async findAll() {
+    const countries: Array<CountryInterface> =
+      await this.countryService.findAll();
+    return countries;
+  }
+
+  @Get(':id')
+  async find(@Param('id') id: string): Promise<CountryInterface | null> {
+    return await this.countryService.findOne(+id);
+  }
+
+  @Post()
+  async create(
```

Comment:
Same note about proper errors, also try to have a unified return type which usually is the DTO.